pym/portage/package/ebuild/config.py

```diff
@@ -2736,6 +2736,8 @@ def environ(self):
 			mydict["ESYSROOT"] = (
 					mydict["SYSROOT"].rstrip(os.path.sep) + self.get("EPREFIX", "")
 					+ os.path.sep)
+		if eapi_attrs.broot and _phase_func_map.get(phase, '').startswith('src_'):
+			mydict["BROOT"] = self.get("EPREFIX", "") + os.path.sep
```

Comment:
I believe disjoint prefix values should be possible now, look at how I implemented it in chewi/portage@6a2f06ec076c47e105292126636d626c63434172, specifically this line in LocationsManager.py. Was there anything wrong with this?

```python
self.broot = portage.const.EPREFIX
```

Reply:
`self.get("EPREFIX", "")` will correspond to the cross-prefix, which is different from portage.const.EPREFIX when cross compiling.

Reply:
The difference is particularly visible in implementation of has_version and best_version in bin/phase-helpers.sh:
```bash
	if ___eapi_has_prefix_variables; then
		# [[ ${root} == / ]] would be ambiguous here,
		# since both prefixes can share root=/ while
		# having different EPREFIX offsets.
		if ${host_root} ; then
			eroot=${root%/}${PORTAGE_OVERRIDE_EPREFIX}/
		else
			eroot=${root%/}${EPREFIX}/
		fi
	else
```

Reply:
@zmedico, right but to be clear, are you agreeging that `portage.const.EPREFIX` is the correct value to use in this instance? Also is it better to setup this variable in config.py as @mgorny did or in LocationsManager.py as I did?

Reply:
Yes, `portage.const.EPREFIX` is correct, but I prefer indirect reference via `PORTAGE_OVERRIDE_EPREFIX` due to https://bugs.gentoo.org/643920. It does seem like a good idea to add a broot attribute to LocationsManager. It can refer to `portage.const.EPREFIX` for now, but ultimately I would like to pass it in via the the constructor in order to solve https://bugs.gentoo.org/643920.

Reply:
I did look at passing it via the constructor but `PORTAGE_OVERRIDE_EPREFIX` is currently set more than 300 lines after the constructor is used. It's in the same very long clause but I don't want to move it in case it breaks something. I'll leave it to you.